tensorflow/core/kernels/attention_ops.cc

```diff
@@ -98,13 +108,13 @@ class ExtractGlimpseOp : public OpKernel {
         context->eigen_cpu_device()) =
         Eigen::ExtractGlimpses(input.tensor<float, 4>().swap_layout(),
                                output_width, output_height, offset_vec,
-                               normalized_, centered_, uniform_noise_);
+                               normalized_, centered_, noise_);
   }
 
  private:
   bool normalized_;
   bool centered_;
-  bool uniform_noise_;
+  string noise_;
```

Comment:
Can you create an enum to encode the valid types ? String comparisons are expensive, and don't run well on GPU.

Reply:
@benoitsteiner Thanks. Done.